eslint/README.md

```diff
@@ -0,0 +1,105 @@
+# [ESL](../../../) ESL migration support
+
+Version: *4.12.0*
+
+Authors: *Natalia Smirnova, Alexey Stsefanovich (ala'n)*.
+
+<a name="intro"></a>
+
+This article aims to assist the smooth migration process from older versions of ESL to the latest version. To support this transition, we have developed an ESLint plugin designed specifically for ESL library. This plugin targets deprecated features and aliases within the library, offering a seamless means of identifying these outdated elements. Additionally, it suggests suitable replacements, thereby ensuring a hassle-free upgrade to the newest version of ESL.
+
+### Installation
+**Note**: Before installing the plugin, ensure that you have the ESLint package of version 8.0.0 or higher. If you do not intend to install ESLint, this article may not be helpful.
+
+To use custom ESLint plugin, you need to install it as npm package:
+
+```bash
+npm install --save-dev @exadel/eslint-plugin-esl
+```
+
+Once installed, the plugin needs to be added in eslint configuration file:
+
+```json
+{
+  // ...
+  "plugins": [
+    "@exadel/esl"
+  ]
+  // ...
+}
+```
+
+Or in YAML:
+```js
+  plugins: 
+    - "@exadel/esl"
+```
+
+### Configuration
+You have the option to utilize the plugin's preset configurations. To keep consistent with esl updates but in light mode - apply 4.0 preset, which is just warns you if you have any deprecated features in your code:
+
+```json
+{
+  // ...
+  "extends": [
+    "plugin:@exadel/esl/default-4"
+  ]
+  // ...
+}
+```
+
+ In order to be compatible with version 5.0.0 (which is coming by the end of this year) - apply 5.0 preset which is strictly deprecates the usage of the features that are going to be removed:
+
+```json
+{
+  // ...
+  "extends": [
+    "plugin:@exadel/esl/default-5"
+  ]
+  // ...
+}
+```
```

Comment:
Please mention the fact that all of the rules are auto correctable so user can use fix option of ESLint to perform batch or automatic update of the code